From 6618b8244c3fa711e0aa06d91da78bb0de06ab0f Mon Sep 17 00:00:00 2001
From: ZhaoLong Wang <wangzhaolong1@huawei.com>
Date: Sat, 5 Nov 2022 10:33:36 +0800
Subject: [PATCH 594/595] [Huawei] ubi: Reserve sufficient buffer length for
 the input mask

Offering: HULK
hulk inclusion
category: feature
bugzilla: 187656

---------------------------------

Because the mask received by the emulate_failures interface
is a 32-bit unsigned integer, ensure that there is sufficient
buffer length to receive and display this value.

Signed-off-by: ZhaoLong Wang <wangzhaolong1@huawei.com>
---
 drivers/mtd/ubi/debug.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/mtd/ubi/debug.c b/drivers/mtd/ubi/debug.c
index 7c5c962142c0..c24fd1dae458 100644
--- a/drivers/mtd/ubi/debug.c
+++ b/drivers/mtd/ubi/debug.c
@@ -333,7 +333,7 @@ static ssize_t dfs_file_read(struct file *file, char __user *user_buf,
 	struct dentry *dent = file->f_path.dentry;
 	struct ubi_device *ubi;
 	struct ubi_debug_info *d;
-	char buf[8];
+	char buf[12];
 	int val;
 
 	ubi = ubi_get_device(ubi_num);
@@ -382,7 +382,7 @@ static ssize_t dfs_file_write(struct file *file, const char __user *user_buf,
 	struct ubi_device *ubi;
 	struct ubi_debug_info *d;
 	size_t buf_size;
-	char buf[8] = {0};
+	char buf[14] = {0};
 	int val;
 
 	ubi = ubi_get_device(ubi_num);
-- 
2.17.1

